src/workclaimer.js

```diff
@@ -0,0 +1,140 @@
+let assert  = require('assert');
+let _       = require('lodash');
+let Promise = require('promise');
+let events  = require('events');
+
+/**
+ * HintPoller polls for hints for pending tasks.
+ *
+ * The azure queues don't know if a task is pending they just store hints of
+ * pending tasks. To be understood this way:
+ *  A) If a task is pending, there is a hint of the task in an azure queue,
+ *  B) If there is an hint in an azure queue, it may or may not be pending.
+ *
+ * It's an if, but not an only-if (think over-approximation).
```

Comment:
> Should we also describe the case where there's a pending task but it does not have a hint -- or do we just consider that a dropped task?

A pending task without a hint cannot have been successfully created.
`queue.createTask` won't return 200 unless the pending hint was en-queued. And if it fails with a random error or crash, the client should retry until it gets 200, or client will know for sure that the task wasn't successfully created.

Note: tasks that aren't successfully created, are typically resolved by deadline expiration...
But we don't really make any promises... I think that's fair.